store/tikv/mocktikv/cop_handler_dag.go

```diff
@@ -370,6 +406,98 @@ func flagsToStatementContext(flags uint64) *stmtctx.StatementContext {
 	return sc
 }
 
+// mockClientStream implements grpc ClientStream interface, its methods are never called.
+type mockClientStream struct{}
+
+func (mockClientStream) Header() (metadata.MD, error) { return nil, nil }
+func (mockClientStream) Trailer() metadata.MD         { return nil }
+func (mockClientStream) CloseSend() error             { return nil }
+func (mockClientStream) Context() goctx.Context       { return nil }
+func (mockClientStream) SendMsg(m interface{}) error  { return nil }
+func (mockClientStream) RecvMsg(m interface{}) error  { return nil }
+
+type mockCopStreamClient struct {
+	mockClientStream
+
+	req      *tipb.DAGRequest
+	exec     executor
+	goCtx    goctx.Context
+	finished bool
+}
+
+type mockCopStreamErrClient struct {
+	mockClientStream
+
+	*errorpb.Error
+}
+
+func (mock *mockCopStreamErrClient) Recv() (*coprocessor.Response, error) {
+	return &coprocessor.Response{
+		RegionError: mock.Error,
+	}, nil
+}
+
+func (mock *mockCopStreamClient) Recv() (*coprocessor.Response, error) {
+	if mock.finished {
+		return nil, io.EOF
+	}
+
+	var resp coprocessor.Response
+	chunk, finish, err := mock.readBlockFromExecutor()
+	if err != nil {
+		if locked, ok := errors.Cause(err).(*ErrLocked); ok {
+			resp.Locked = &kvrpcpb.LockInfo{
+				Key:         locked.Key,
+				PrimaryLock: locked.Primary,
+				LockVersion: locked.StartTS,
+				LockTtl:     locked.TTL,
+			}
+		} else {
+			resp.OtherError = err.Error()
+		}
+		return &resp, nil
+	}
+	if finish {
+		// Just mark it, need to handle the last chunk.
+		mock.finished = true
+	}
+
+	data, err := chunk.Marshal()
```

Comment:
Can we return chuck directly like `SelectResponse`? Or we can return chuck.RowsData. Then we can reduce the time of `Marshal`.

Reply:
No, we can't change the data format.
This data format is the result after discussing with TiKV team, mock tikv keep consistent to TiKV.